### ISSUE TITLE
What does "Ijk_MapFail" mean?

### ISSUE THREAD

**Issue by mspi21** [angr/angr#4714]
### Question

I have been trying to use *angr* to solve a reverse engineering crackme with heavily obfuscated control flow that takes a password on the stdin, does God knows what to it (manual static analysis is not an option) and then prints either a success or failure message to the stdout. (I am not attaching the binary because I'm not sure I have the rights to "redistribute" it, but if it should help, I'll figure something out.)

I have run into an issue where my simulation manager produces a single errored state &mdash; printing it produces `<State errored with "Cannot execute following jumpkind Ijk_MapFail">`, but there is no address or any other indication of what might have gone wrong, whereas with other kinds of errors (such as instruction decoding errors), the address was provided, and I could patch/hook the instruction at that place. I have tried looking through the documentation and source code to understand what exactly `Ijk_MapFail` means, but I feel like figuring it out by myself is beyond my limits.

To me, a "MapFail jumpkind" sounds like maybe the branch target address lies outside of any mapped virtual memory region &mdash; in that case, ~~it would be helpful to at least be able to retrieve the address of the failing instruction~~ I have figured out how to do this. (It seems the instruction is completely malformed and probably unreachable in the program.)

I will appreciate any help, resources and/or tips on this error message in particular, and/or on dealing with non-standard binaries with heavily obfuscated control flow using *angr* in general.

**Comment by rhelmot** [comment]
From what I can tell, MapFail is only triggered on x86 segmentation producing a bad address. Are your segment tables set up correctly?

**Comment by mspi21** [comment]
> From what I can tell, MapFail is only triggered on x86 segmentation producing a bad address. Are your segment tables set up correctly?

You're right, I found out that the disassembled instruction that caused the "MapFail" was this one:
```s
jmp     far ptr 6631h:291F67AFh
```
where segment number 0x6631 obviously doesn't make sense. I'm convinced this instruction was put in the binary for this sole purpose &mdash; to make symbolic execution harder &mdash; or maybe it overlaps with another, more reasonable instruction. In any case, I've added the address to the "avoid" list.

Thank you for your insight, I think this answers my question.
